rust/hyperlane-base/src/metrics/core.rs

```diff
@@ -468,3 +472,94 @@ impl Debug for CoreMetrics {
         )
     }
 }
+
+#[derive(Debug, Eq, PartialEq, Hash)]
+struct AppContextKey {
+    origin: HyperlaneDomain,
+    destination: HyperlaneDomain,
+    app_context: String,
```

Comment:
it's supposed to be a friendly label that relates to either the message recipient or the default ISM. So I expect this to be something like "neutron_mantapacific_tia" (for tracking that warp route, regardless if the ISM gets switched out), or "default_ism", etc

Reply:
hm, then `app_context` make sense because it's so general. Could only think of `app_validator_group`, although it wouldn't be obvious that `group` means "a union of validator sets"

Reply:
sg, I'll just keep as app_context and we can always rename